docs/docsite/rst/dev_guide/developer_labs.rst

```diff
@@ -0,0 +1,136 @@
+Developer Labs
+==============
+
+.. contents:: Topics
+
+Ansible has a lot of documentation and is increasingly building out development docs, but not everyone learns the same way. This page provides self-guided labs which are intended to teach new Ansible developers the ins and outs of writing code and working with the github PR process and shippable CI system. It is expected that the developer does a commit+push and waits for tests to run after each step in the process to learn the behavior of shippable tests and ansibot. You are expected to see a lot of errors and failures through this process. Please use the docs, the sourcecode, the community and the Ansible core team (in that order) if you find yourself stuck at any point.
+
+.. _level_1:
+
+Level 1
+-------
+
+.. _module:
+
+Module
+++++++
+
+Build a module and tests from scratch by starting with a broken python script.
+
+.. _module_hello_world:
+
+module from scratch
+```````````````````
+
+1. fork ansible
```

Comment:
1b enable shippable on your repo 

should we <link to instructions> or just add them here?